plugins/aggregators/histogram/histogram.go

```diff
@@ -160,8 +183,18 @@ func (h *HistogramAggregator) Add(in telegraf.Metric) {
 // Push returns histogram values for metrics
 func (h *HistogramAggregator) Push(acc telegraf.Accumulator) {
 	metricsWithGroupedFields := []groupedByCountFields{}
+	now := time.Time{}
 
-	for _, aggregate := range h.cache {
+	if h.ExpirationInterval != 0 {
+		now = timeNow()
+	}
+
+	for id, aggregate := range h.cache {
+		if !now.IsZero() && now.After(aggregate.expireTime) {
```

Comment:
With the change above do
```suggestion
		if h.ExpirationInterval != 0 && now.After(aggregate.expireTime) {
```